cmd/maestroctl/cmd/plan/planList.go

```diff
@@ -0,0 +1,159 @@
+package plan
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/xlab/treeprint"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"os"
+	"os/user"
+
+	"path/filepath"
+
+	maestrov1alpha1 "github.com/maestrosdk/maestro/pkg/apis/maestro/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+)
+
+var (
+	instance   string
+	kubeConfig string
+	namespace  string
+)
+
+const (
+	defaultConfigPath = ".kube/config"
+)
+
+func NewPlanListCmd() *cobra.Command {
+	listCmd := &cobra.Command{
+		//Args: cobra.ExactArgs(1),
+		Use:   "list",
+		Short: "Lists all available plans of a particular instance.",
+		Long: `
+	# View plan status
+	maestroctl plan list --instance=<instanceName>`,
+		Run: planListCmd,
+	}
+
+	listCmd.Flags().StringVar(&instance, "instance", "", "The instance name.")
+	listCmd.Flags().StringVar(&kubeConfig, "kubeconfig", "", "The file path to kubernetes configuration file; defaults to $HOME/.kube/config")
+	listCmd.Flags().StringVar(&namespace, "namespace", "default", "The namespace where the operator watches for changes.")
+
+	return listCmd
+}
+
+func planListCmd(cmd *cobra.Command, args []string) {
+
+	instanceFlag, err := cmd.Flags().GetString("instance")
+	if err != nil || instanceFlag == "" {
+		log.Printf("Error: No instance was provided")
+		return
+	}
+
+	mustKubeConfig()
+
+	_, err = cmd.Flags().GetString("kubeconfig")
+	if err != nil || instanceFlag == "" {
+		log.Printf("Flag Error: %v", err)
+	}
+
+	p, err := planList()
+	if err != nil {
+		log.Printf("Error: %v", err)
+	}
+	tree := treeprint.New()
+
+	for _, plan := range p {
+		tree.AddBranch(plan)
+	}
+	fmt.Printf("List of current plans for \"%s\" in namespace \"%s\":\n", instance, namespace)
+	fmt.Println(tree.String())
+
+}
+
+func planList() ([]string, error) {
+
+	planList := make([]string, 0)
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	//  Create a Dynamic Client to interface with CRDs.
+	dynamicClient, err := dynamic.NewForConfig(config)
```

Comment:
@gerred What's the best way to generate the client so we don't have to use the Dynamic client for this?